sslib/allpatch.py

```diff
@@ -201,9 +223,134 @@ def patch_arc_replacements(self):
+
+        if os.path.isdir(dataPath / "AdditionalArcs"):
+            for arcPath in (dataPath / "AdditionalArcs").glob("*.arc"):
+                arcName = arcPath.parts[-1]
+                if arcName == "Alink.arc" or arcName == "Bird_Link.arc":
+                    continue  # ignore arcs that get patched separately
+                self.arc_replacements[arcName] = arcPath
+
+        #### loftwing patches ####
+        if self.current_loftwing_model_pack_name == "Default":
+            dataPath = LINK_MODEL_DATA_PATH / "Loftwing"
+            arcPath = OARC_PATH / "Bird_Link.arc"
+        else:
+            dataPath = (
+                CUSTOM_MODELS_PATH / self.current_loftwing_model_pack_name / "Loftwing"
+            )
+            arcPath = dataPath / "Bird_Link.arc"
+
+        metaDataPath = (
+            CUSTOM_MODELS_PATH
+            / self.current_loftwing_model_pack_name
+            / "Loftwing"
+            / "metadata.json"
+        )
+        if os.path.isfile(metaDataPath):
+            with open(metaDataPath) as f:
+                metaData = json.load(f)
+        else:
+            metaData = None
+        birdArcBytes = arcPath.read_bytes()
+        parsedBirdArc = U8File.parse_u8(BytesIO(birdArcBytes))
+
+        masksPath = dataPath / "Masks"
+        if os.path.isdir(masksPath) and metaData.get("Colors"):
+            parsedBirdArc = self.do_texture_recolour(
+                parsedBirdArc, masksPath, colorData=metaData["Colors"]
+            )
+
+        (MODIFIED_ARC_TEMP_PATH / "Bird_Link.arc").write_bytes(
+            parsedBirdArc.to_buffer()
+        )
+        self.arc_replacements["Bird_Link.arc"] = (
+            MODIFIED_ARC_TEMP_PATH / "Bird_Link.arc"
+        )
+
+    def do_texture_recolour(
+        self, arcData: U8File, maskFolderPath: Path, colorData: dict
+    ) -> U8File:
+        maskLookup = {}
+        for p in maskFolderPath.iterdir():
+            if match := MASK_REGEX.match(str(p)):
+                if match.group("texName") not in maskLookup:
+                    maskLookup[match.group("texName")] = []
+                maskLookup[match.group("texName")].append(match.group("colorGroupName"))
+
+        brresData = arcData.get_file_data("g3d/model.brres")
+        parsedBRRES = BRRES.parse_brres(BytesIO(brresData))
+
+        for texName in maskLookup:
+            texPath = f"Textures(NW4R)/{texName}"
+            imageData: np.array = parsedBRRES.get_file_data(path=texPath)
+            maskPaths = []
+            colors = []
+            process = False
+            for colorGroup in maskLookup[texName]:
+                if colorGroup not in colorData:
+                    print(f"{colorGroup} not in metadata")
+                    continue
+                if colorData[colorGroup] == "Default":
+                    continue
+                maskPath = str(
+                    maskFolderPath / (str(texName) + "__" + str(colorGroup) + ".png")
```

Comment:
```suggestion
                    maskFolderPath / f"{texName}__{colorGroup}.png")
```
Prefer f-strings (assuming that works here)